app/src/main/java/com/example/dashcarr/presentation/tabs/settings/SettingsFragment.kt

```diff
@@ -13,15 +17,35 @@ class SettingsFragment : BaseFragment<FragmentSettingsBinding>(
     private val viewModel: SettingsViewModel by viewModels()
 
     override fun observeViewModel() {
-        TODO("Not yet implemented")
+        viewModel.logOutState.collectWithLifecycle(viewLifecycleOwner) {
+            findNavController().navigate(R.id.action_global_loginFragment)
+        }
     }
 
     override fun initListeners() {
-        TODO("Not yet implemented")
+        binding.btnLogout.setOnClickListener {
+            viewModel.logOut()
+        }
+        binding.btnProfileSettings.setOnClickListener {
+            //findNavController().navigate(R.id.)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initListeners()
+        observeViewModel()
     }
 
+
+
```

Comment:
Just a small tip: Android Studio supports "Actions on Save" actions
<img width="707" alt="image" src="https://github.com/StianJohnsen/dte2503_team_3_repo/assets/57566364/fdaaf12a-c610-480b-a636-2230982fc4be">
It removes i.e. unnecessary breaks/spaces

Reply:
show me where is this preferences